src/tabwidget.cpp

```diff
@@ -352,26 +353,66 @@ void TabWidget::removeCurrentTab()
 
 int TabWidget::switchToRight()
 {
-    int next_pos = currentIndex() + 1;
-    if (next_pos < count())
-        setCurrentIndex(next_pos);
-    else
-        setCurrentIndex(0);
-    findParent<MainWindow>(this)->updateDisabledActions();
+    switchTab((currentIndex() + 1) % count());
     return currentIndex();
 }
 
 int TabWidget::switchToLeft()
 {
-    int previous_pos = currentIndex() - 1;
-    if (previous_pos < 0)
-        setCurrentIndex(count() - 1);
-    else
-        setCurrentIndex(previous_pos);
-    findParent<MainWindow>(this)->updateDisabledActions();
+    switchTab(currentIndex() ? currentIndex() - 1 : count() - 1);
     return currentIndex();
 }
 
+void TabWidget::switchTo1()
+{
+    switchTab(0);
+}
+
+void TabWidget::switchTo2()
+{
+    switchTab(1);
+}
+
+void TabWidget::switchTo3()
+{
+    switchTab(2);
+}
+
+void TabWidget::switchTo4()
+{
+    switchTab(3);
+}
+
+void TabWidget::switchTo5()
+{
+    switchTab(4);
+}
+
+void TabWidget::switchTo6()
+{
+    switchTab(5);
+}
+
+void TabWidget::switchTo7()
+{
+    switchTab(6);
+}
+
+void TabWidget::switchTo8()
+{
+    switchTab(7);
+}
+
+void TabWidget::switchTo9()
+{
+    switchTab(8);
+}
+
+void TabWidget::switchTo10()
+{
+    switchTab(9);
+}
+
```

Comment:
I guess these function are not necessary if 10 setup_Action calls are transformed into a loop.